server/shared/types/terms/term.ts

```diff
@@ -31,20 +31,19 @@ export type TermValues = {
 	[key: string | number]: BaseValue
 }
 
-// THIS IS WRONG!!!!
-export type ValuesGroup = {
-	type: 'values'
-	values: { key: number | string; label: string }[]
-	// color?: string
+export type GroupEntry = {
+	name: string
+	/** `values` is the default. if `values`, values must be present
+	 * if `filter`, filter must be present */
+	type: 'values' | 'filter'
 }
 
-export type FilterGroup = {
-	type: 'filter'
-	filter: Filter
+export type ValuesGroup = GroupEntry & {
+	values: { key: number | string; label: string }[]
 }
 
-export type GroupEntry = (ValuesGroup | FilterGroup) & {
-	name: string
+export type FilterGroup = GroupEntry & {
+	filter: Filter
 }
```

Comment:
The following seems simpler. 

Also note that FiltersGroup may already be deprecated/not used, need to verify if it's still needed.

```js
export type ValuesGroup = {
	type: 'values'
	values: { 
		key: number | string
		label: string
		dt?: number
	}[]
}

export type FilterGroup = {
	name: string
	type: 'filter'
	filter: Filter
}

export type GroupEntry = ValuesGroup | FilterGroup

```